recipe/meta.yaml

```diff
@@ -1,31 +1,32 @@
 {% set name = "ctypesgen" %}
-{% set version = "1.1.2.dev4+g8a62d02" %}
+{% set version = "1.1.1a" %}
 
 package:
-  name: {{ name|lower }}
+  name: {{ name|lower }}-pypdfium2-team
   version: {{ version }}
```

Comment:
In general, we should avoid having versions that the upstream maintainer could potentially generate in the future; also version suffixes that contain `a` and `b` may run afoul of [PEP-440 Pre-Releases](https://peps.python.org/pep-0440/#pre-releases) rules and lead to conda thinking this is an alpha or beta release. (We have special carve outs in the solver for `openssl` and `libjpeg` for exactly this reason.)

The pseudo-canonical way of handling builds from an arbitrary git commit is to use the `{last_tag}+{number of commits}_g{shortened commit id}`. The way to auto-generate that is to use `version: {{ GIT_DESCRIBE_TAG }}+{{ GIT_BUILD_STR }}`; see the [conda nightlies recipe](https://github.com/conda/conda/blob/6d5f755/recipe/meta.yaml#L3) for an example.

Reply:
thanks for all the useful info!

Reply:
By the way, I agree that

> we should avoid having versions that the upstream maintainer could potentially generate in the future

But in this case the maintainer may introduce breaking changes in the API on the master branch of his fork (see [comments](https://github.com/AnacondaRecipes/pypdfium2-feedstock/commit/cd045f19c634da481a290e8b1226706b2d5d92d4#r124139490)), so I thought it is safer to do it this way. And by doing it this way, I don't see a better way of differentiating this package from the [upstream one](https://anaconda.org/anaconda/ctypesgen).

Reply:
By the way, it seems that if I write the `package` section like so:
> package:
  name: {{ name|lower }}-pypdfium2-team
  version: {{ GIT_DESCRIBE_TAG }}+{{ GIT_BUILD_STR }}
  
  I get error:
> Error: git is not installed in your root environment or as a build requirement.

I get that on linux where `git` is by default not present.

Reply:
> But in this case the maintainer may introduce breaking changes in the API on the master branch of his fork

Sorry...that was my mistake in not communicating clearly. When I said "versions that the upstream maintainer", I meant the `pypdfium2-team/ctypesgen` maintainers, and not the upstream of the upstream `ctypesgen/ctypesgen` maintainers.

> differentiating this package from the upstream one.

IMO, the package name (`ctypesgen-pypdfium2-team`) is probably the best way to do it.

Reply:
> > Error: git is not installed in your root environment or as a build requirement.
>
> I get that on linux where `git` is by default not present.

Looking in [the `conda-build` source](https://github.com/conda/conda-build/blob/3.24.0/conda_build/source.py#L330-L332), I think "or as a build requirement" is an erroneous error message, as `conda-build` doesn't _actually_ look into the `build` prefix to find `git`. It only looks into the base environment and whatever directories are currently in your `$PATH`.

Reply:
as discussed, I'll exceptionally formulate the version manually for this release until a `git` distribution is guaranteed on our CI workers.